src/server/modules/user/index.js

```diff
@@ -65,6 +66,60 @@ if (settings.user.auth.facebook.enabled) {
   );
 }
 
+if (settings.user.auth.google.enabled) {
+  passport.use(
+    new GoogleStrategy(
+      {
+        clientID: settings.user.auth.google.clientID,
+        clientSecret: settings.user.auth.google.clientSecret,
+        callbackURL: '/auth/google/callback'
+      },
+      async function(accessToken, refreshToken, profile, cb) {
+        const { id, username, displayName, emails: [{ value }] } = profile;
+        try {
+          let user = await User.getUserByGoogleIdOrEmail(id, value);
+
+          if (!user) {
+            const isActive = true;
+            const [createdUserId] = await User.register({
+              username: username ? username : value,
+              email: value,
+              password: id,
+              isActive
+            });
+
+            await User.createGoogleOuth({
+              id,
+              displayName,
+              userId: createdUserId
+            });
+
+            await User.editUserProfile({
+              id: createdUserId,
+              profile: {
+                firstName: profile.name.givenName,
+                lastName: profile.name.familyName
+              }
+            });
```

Comment:
Setting the local user profile from the google profile information.

Should we do the same for Facebook?